ui/src/Model.elm

```diff
@@ -33,53 +32,52 @@ init location =
         updatedModel =
             routeFromUrl defaultModel location
     in
-        updatedModel ! [ getBuildRecordList updatedModel ]
+        updatedModel
+            ! [ getFilters "product"
+              , getFilters "channel"
+              , getFilters "platform"
+              , getFilters "version"
+              , getFilters "locale"
+              , getBuildRecordList updatedModel
+              ]
+
+
+getFilters : String -> Cmd Msg
+getFilters filterName =
+    client
+        |> Kinto.getList (filterRecordResource filterName)
+        |> Kinto.sortBy [ "id" ]
+        |> Kinto.send (FiltersReceived filterName)
 
 
 getBuildRecordList : Model -> Cmd Msg
 getBuildRecordList { buildIdFilter, productFilter, channelFilter, platformFilter, versionFilter, localeFilter } =
+    {- FIXME: https://github.com/Kinto/kinto/issues/1217: here we surround all qs param values with quotes
+       in case they're treated as numbers by Kinto, which makes it crash.
+    -}
     let
-        request =
-            client
-                |> Kinto.getList recordResource
-                |> Kinto.limit pageSize
-                |> Kinto.sortBy [ "-build.date" ]
+        applyListFilter apply filter request =
```

Comment:
> Also, why List in the `applyListFilter` name?

Because we're using `getList` just above, but I can totally update the name to `applyFilter`. I'll do that in another PR though, as this one has landed already.